### PR TITLE
tensorflow: add 0.12.0

### PR DESCRIPTION
**Pull request by sergei-mironov**
###### Motivation for this change
Bring popular machine learning framework up to date. Make it compatible with Python3 packages.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -1025,6 +1025,26 @@ in modules // {
     };
   };
 
+  funcsigs1 = buildPythonPackage rec {
+    name = "funcsigs-1.0.0";
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: We already have funcsigs 1.0.2, no need for adding 1.0.0

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -13733,6 +13753,28 @@ in modules // {
     };
   });
 
+  mock20 = buildPythonPackage (rec {
+    name = "mock-2.0.0";
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: we already have mock 2.0

**Review by FRidh — CHANGES_REQUESTED** [review]
Duplicate of #20794

**Comment by teh** [comment]
1.0 was released as well, so maybe we can update: https://github.com/tensorflow/tensorflow/releases/tag/v1.0.0

**Comment by jyp** [comment]
Obsoleted by #23040, please close.

**Pull request closed without merging**